go/tools/builders/generate_test_main.go

```diff
@@ -62,13 +62,15 @@ type Cases struct {
 
 const testMainTpl = `
 package main
+// This must be first.
```

Comment:
Need a more complete comment:

```
// This package must be initialized before packages being tested.
```

Also, I don't think package initialization order is specified. This seems to work, but I don't know if it's guaranteed to work.

```
// NOTE: this relies on the order of package initialization, which is undefined
// behavior. The spec only guarantees that imported packages are initialized
// before their importers.
```

For reference: https://golang.org/ref/spec#Package_initialization

Reply:
The last sentence _seems_ to imply that it is guaranteed--

> To ensure reproducible initialization behavior, build systems are encouraged to present multiple files belonging to the same package in lexical file name order to a compiler.

But I agree, it's not super clear.

Reply:
I asked some coworkers about this, and they pointed me to https://golang.org/issue/31636. From that conversation, it looks like the order is indeed unspecified, but they decided not to make it random. It may be changed (and specified) in the future.

Reply:
Ah, very cool, thanks.